client/my-sites/posts/post-total-views.jsx

```diff
@@ -4,68 +4,67 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import StatUpdateIndicator from 'components/stat-update-indicator';
 import Gridicon from 'components/gridicon';
-import sitesList from 'lib/sites-list';
 import QueryPostStats from 'components/data/query-post-stats';
+import QuerySites from 'components/data/query-sites';
 import { getPostStat } from 'state/stats/posts/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
 
-const sites = sitesList();
+function PostTotalViews( { clickHandler, numberFormat, post, slug, translate, viewCount } ) {
+	const postId = post.ID,
+		siteId = post.site_ID;
+	let viewsCountDisplay = '',
+		viewsTitle;
 
-var PostTotalViews = React.createClass( {
-
-	propTypes: {
-		post: PropTypes.object.isRequired,
-		viewCount: PropTypes.number,
-		clickHandler: PropTypes.func
-	},
-
-	render() {
-		const { viewCount, post } = this.props,
-			postId = post.ID,
-			siteId = post.site_ID;
-		let viewsCountDisplay = '',
-			viewsTitle;
-
-		if ( viewCount && ! isNaN( viewCount ) ) {
-			viewsCountDisplay = this.numberFormat( viewCount );
-			viewsTitle = this.translate( '1 Total View', '%(count)s Total Views', {
-				count: viewCount,
-				args: {
-					count: viewCount
-				}
-			} );
-		} else {
-			viewsTitle = this.translate( 'Total Views' );
-		}
+	if ( viewCount && ! isNaN( viewCount ) ) {
+		viewsCountDisplay = numberFormat( viewCount );
+		viewsTitle = translate( '1 Total View', '%(count)s Total Views', {
+			count: viewCount,
+			args: {
+				count: viewCount
+			}
+		} );
+	} else {
+		viewsTitle = translate( 'Total Views' );
+	}
 
-		const site = sites.getSite( siteId );
+	return (
+		<a href={ `/stats/post/${postId}/${slug}` }
+			className={ classNames( {
+				'post__total-views': true,
+				'is-empty': ! viewsCountDisplay
+			} ) }
+			title={ viewsTitle }
+			onClick={ clickHandler }>
+			<QueryPostStats siteId= { siteId } postId={ postId } stat="views" />
+			<QuerySites siteId={ siteId } />
```

Comment:
cc @aduth would we ever have a case where the current post siteId isn't the currently selected site?